[export] do not unlift cond/map submodules

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118487
* #118486
* __->__ #118485
* #118484
* #118505
* #118483

I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

suo committed:
[export] do not unlift cond/map submodules

I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

zhxchen17 commented:
> What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

I agree with this contract (for now lol). IMO even unlifting itself should be an implementation detail at the current moment because .module() technically should just return something "acting like" the original eager module in many ways. Unlifting is one approach to this.

@tugsbayasgalan I remember you mainly worked on unlifting, do you remember any context on this?

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]

suo committed:
Update on "[export] do not unlift cond/map submodules"


I don't think we should be unlifting HOO submodules.

What is the constract of unlifting? It is: restore the original calling convention of the module, undoing the transformation in which we lift parameters, buffers, and constants to inputs in the graph.

Unlifting does *not* make any guarantees about what's going on inside the module. It's still a flat module. So why should we lift the cond/map submodules? It doesn't have anything to do with the contract stated above; it's some internal stuff that doesn't affect how the module will be called.

Further, this code as written modifies the state dict; adding a new buffer that is actually duplicate of a previous buffer. Modifying the state dict from the original eager module is never correct.

Differential Revision: [D53160713](https://our.internmc.facebook.com/intern/diff/D53160713/)

[ghstack-poisoned]
